### PR TITLE
feat(prospecting): [BACK-1338] Handle prospect pre-existing as a curated item

### PR DESCRIPTION
**Pull request by nina-py**
## Goal

Update the workflow for prospects that already exist in the corpus so that curators can go straight to scheduling these instead of skipping through the edit screens. For any manually added URLs, the code will skip through the edit screen as defined initially in BACK-1338 - please see the two video walkthroughs below.

## Reference

Tickets:

- https://getpocket.atlassian.net/browse/BACK-1338

### Updated workflow:
- Blue notice for manual additions that are already in the corpus
- red brick pill for prospects already in the corpus
- scheduled surface dropdowns get locked/unlocked as required (manual additions are "free", user can choose any available tab, on other occasions the dropdown is locked to the surface the user is prospecting for).

https://user-images.githubusercontent.com/22447785/161965446-62eea9dc-9f1b-4eca-83d5-997d66f5e1a6.mov

----> Outdated video walkthroughs below

### Main workflow

https://user-images.githubusercontent.com/22447785/160321779-76bf6b45-d0aa-4256-9d06-5e14f220f03d.mov

#### Manual add workflow

https://user-images.githubusercontent.com/22447785/160336915-cb86be87-d218-4c27-9653-02b5d810d01b.mov

### CODE REVIEW

File: src/curated-corpus/components/AddProspectForm/AddProspectForm.tsx [hunk 1]
```diff
@@ -61,15 +61,15 @@ export const AddProspectForm: React.FC<
...
-        </Grid>
-      )}
+        {formik.isSubmitting && (
+          <Grid item xs={12}>
+            <Box mb={3}>
+              <LinearProgress />
+            </Box>
+          </Grid>
+        )}
+      </form>
```
Comment (Herraj) [inline src/curated-corpus/components/AddProspectForm/AddProspectForm.tsx:72]: Thanks for adding this!

File: src/curated-corpus/components/ExistingProspectCard/ExistingProspectCard.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+          >
+            <span>{item.publisher}</span>
+          </Typography>{' '}
+          <Box py={1}>
+            <Chip
+              variant="outlined"
+              color="primary"
+              label={item.language}
+              icon={<LanguageIcon />}
+            />{' '}
```
Comment (Herraj) [inline src/curated-corpus/components/ExistingProspectCard/ExistingProspectCard.tsx:71]: These annoying spaces snuck in again

File: src/curated-corpus/pages/ProspectingPage/ProspectingPage.tsx [hunk 3]
```diff
@@ -615,6 +632,19 @@ export const ProspectingPage: React.FC = (): JSX.Element => {
 
           {prospects &&
             prospects.map((prospect) => {
+              if (prospect.approvedCorpusItem) {
```
Comment (Herraj) [inline src/curated-corpus/pages/ProspectingPage/ProspectingPage.tsx:635]: nit pick: can we do this like we have it above

`prospect.approvedCorpusItem && <ExistingProspectCard ...`

**Review by Herraj — APPROVED** [review]
Good stuff! :shipit:

**Review by jpetto — CHANGES_REQUESTED** [review]
this is looking great! a couple requests:

- for both prospects coming in from prospect API and items manually added, we need to inform the curator that they are already in the corpus. this might be an awkward bit of UI :/
- for prospects from prospect API that are already in the corpus, the schedule form should have the current scheduled surface pre-selected. we don't need to do this for manually added items.

File: src/curated-corpus/components/AddProspectForm/AddProspectForm.tsx (hunk 1 above)
Comment (nina-py) [reply src/curated-corpus/components/AddProspectForm/AddProspectForm.tsx:72]: I wish it worked though! :)

File: src/curated-corpus/components/ExistingProspectCard/ExistingProspectCard.tsx (hunk 2 above)
Comment (nina-py) [reply src/curated-corpus/components/ExistingProspectCard/ExistingProspectCard.tsx:71]: This one is deliberately there to make some space between the two tags, but the one above snuck in - removed.

File: src/curated-corpus/pages/ProspectingPage/ProspectingPage.tsx (hunk 3 above)
Comment (nina-py) [reply src/curated-corpus/pages/ProspectingPage/ProspectingPage.tsx:635]: Pairing - decided to keep this as is

File: src/curated-corpus/pages/ProspectingPage/ProspectingPage.tsx [hunk 4]
```diff
@@ -391,20 +403,25 @@ export const ProspectingPage: React.FC = (): JSX.Element => {
    * common stuff we (may) need to do after creating an approved item
    *
    * @param approvedItem
-   * @param filterProspects (boolean): whether or not we need to filter the list of prospects on the screen
+   * @param isRegularProspect
+   * and also
```
Comment (Herraj) [inline src/curated-corpus/pages/ProspectingPage/ProspectingPage.tsx:407]: nit pick: seems like the comment got cut off

**Review by Herraj — APPROVED** [review]
🚀

**Pull request merged**